### PR TITLE
Show React owner info in Elements panel

### PR DESCRIPTION
**Pull request by bvaughn**
### [Loom overview](https://www.loom.com/share/bd0a0073d138423799187230ec17a299)

https://github.com/replayio/devtools/assets/29597/91b850ec-3763-49df-84ee-e8789168a0e8

### CODE REVIEW

File: packages/replay-next/components/elements-new/suspense/DOMReactCache.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      if (message) {
+        console.error(message);
+      }
+    }
+
+    return [null, null];
+  },
+});
+
+function getReactInfo(domNodeId: ObjectId): Data {
```
Comment (jasonLaster) [inline packages/replay-next/components/elements-new/suspense/DOMReactCache.ts:61]: I wonder if there's a way to flag functions that are meant to be interpolated into an eval? It's a small thing, but it makes reading the code a bit easier.

File: packages/replay-next/components/elements-new/suspense/DOMReactCache.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  ) {
+    const domNode = __RECORD_REPLAY__.getObjectFromProtocolId(domNodeId);
+
+    for (let name in domNode) {
+      // HACK
+      // Although a lot of tooling uses this attribute, it's not meant to be a public API.
+      // We already have a better DOM-node-to-Fiber-id mapping (coming from React DevTools)
+      // but it might not have been instantiated yet and it's in a place the Elements panel can't access.
+      // This approach is a work around with limited perf impact, but it's pretty hacky.
+      if (name.startsWith("__reactFiber$")) {
```
Comment (jasonLaster) [inline packages/replay-next/components/elements-new/suspense/DOMReactCache.ts:75]: Thanks for the comment. It's nice to see that we can leverage the data coming from React DevTools in the future.

File: packages/replay-next/components/elements-new/suspense/DOMReactCache.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        const fiber = domNode[name];
+        if (fiber) {
+          // TRICKY
+          // The Fiber directly associated with any host node (DOM element) will always be the host Fiber (e.g. <div/>)
+          // When we display a rendered-by label, that should point to the "debug owner" (the class/function component that created the <div/>)
+          // React only store "debug owner" information in DEV builds though.
+          // We could fall back to the nearest user component for production builds,
+          // but that would potentially be confusing/misleading so it's probably best to show nothing.
+          const debugOwner = fiber._debugOwner;
```
Comment (jasonLaster) [inline packages/replay-next/components/elements-new/suspense/DOMReactCache.ts:84]: 👍

File: packages/replay-next/components/elements-new/suspense/DOMReactCache.ts (hunk 2 above)
Comment (markerikson) [reply packages/replay-next/components/elements-new/suspense/DOMReactCache.ts:75]: Note that the "React Event Listeners" functionality (both the J2C implementation and the actual "Event Listeners" panel) rely on this same implementation detail of React stuffing data related to the fiber onto the DOM node.

**Pull request merged**